src/hf_hydrodata/data_catalog.py

```diff
@@ -473,58 +412,104 @@ def _get_preferred_catalog_entry(entries: List[dict]) -> dict:
     elif len(entries) == 1:
         result = entries[0]
     else:
-        preferred_file_types = ["pfb", "tif", "netcdf"]
-        ambiguous_entries = []
-        preferred_entry = None
-        preferred_entry_type = None
-        for entry in entries:
-            file_type = entry["file_type"]
-            if file_type in preferred_file_types:
-                if preferred_entry is None:
-                    preferred_entry = entry
-                    preferred_entry_type = file_type
-                else:
-                    if preferred_entry_type == file_type:
-                        ambiguous_entries.append(entry)
-                        ambiguous_entries.append(preferred_entry)
-                        preferred_entry = None
-                    elif preferred_file_types.index(
-                        file_type
-                    ) < preferred_file_types.index(preferred_entry["file_type"]):
-                        preferred_entry = entry
-            else:
-                ambiguous_entries.append(entry)
-        if preferred_entry:
-            result = preferred_entry
-        elif len(ambiguous_entries) == 1:
-            result = ambiguous_entries[0]
-        elif len(ambiguous_entries) > 1:
-            # Try to dis-ambiguate entries using dataset_version
-            max_dataset_version = None
-            preferred_entry = None
-            for ambiguous_entry in ambiguous_entries:
-                dataset_version = ambiguous_entry["dataset_version"]
-                if dataset_version is not None:
-                    if (
-                        max_dataset_version is None
-                        or dataset_version > max_dataset_version
-                    ):
-                        max_dataset_version = dataset_version
-                        preferred_entry = ambiguous_entry
-                    elif dataset_version == max_dataset_version:
-                        raise ValueError(
-                            _ambiguous_error_message(preferred_entry, ambiguous_entry)
-                        )
-            if preferred_entry is not None:
-                result = preferred_entry
-            else:
-                raise ValueError(
-                    _ambiguous_error_message(ambiguous_entries[0], ambiguous_entries[1])
-                )
+        # There is more than one entry that matches the filter so pick the preferred entry
+        preference_states = [
+            {
+                "preference_state_key": "file_type",
+                "preferred_values": ["pfb", "tif", "netcdf"],
+            },
+            {
+                "preference_state_key": "aggregation",
+                "preferred_values": ["mean", "sum", "median", "static", "max", "min"],
+            },
+            {"preference_state_key": "dataset_version", "preferred_values": None},
+        ]
+
+        # evaluate the ambiguous data catalog entries against a preference state to find preferences
+        ambiguous_entries = entries
```

Comment:
Yes this is necessary because the ambiguous_entries list is used in the list to try to reduce the number of ambiguous entries using each different type of column with preferred values in the loop. This statement initialized the list of ambiguous entries to the initial list of catalog entries that matched the users filter.

Reply:
I guess I was thinking that if we aren't using `entries` anymore after this than we don't need a new value for `ambiguous_entries` and if we are using `entires` again, maybe this should be: `ambiguous_entries = entries.copy()`

Reply:
Ok I will change it to ambiguous_entries = entries.copy(). I wanted a new name after I determine there is more than one item in the list, but a copy is a shallow copy and means I do not affect entries. I will change this.

Reply:
I changed the code to ambiguous_entries = entries.copy()